ExternalModule.php

```diff
@@ -9,11 +9,42 @@
 use ExternalModules\AbstractExternalModule;
 use ExternalModules\ExternalModules;
 use REDCap;
+use Logging;
 
 /**
  * ExternalModule class for SuspensionWarning.
  */
 class ExternalModule extends AbstractExternalModule {
+ 	/**
+     * @inheritdoc
+     */
+    function redcap_every_page_before_render($project_id) 
```

Comment:
You don't require authentication.  That's a **big** problem.  The user *must* authenticate before this change is made.